### PR TITLE
bump to zio 2 rc3

### PR DESCRIPTION
**Pull request by jamesward**
I'm not sure if I got all the `ZManaged` -> `Scope` stuff right but it compiles.

### CODE REVIEW

File: experiments/src/main/scala/HelloZio/CalculatorExample.scala [hunk 1]
```diff
@@ -114,15 +114,16 @@ object CalculatorExample extends zio.App:
     String
   ] =
     for
-      (number1, number2) <-
-        ZIO {
+      number <- // note that `(number1, number2)` now results in: value withFilter is not a member of zio.ZIO[Any, Nothing, (Float, Float)], but could be made available as an extension method.
```
Comment (swoogles) [inline experiments/src/main/scala/HelloZio/CalculatorExample.scala:117]: Yep, they talked about how removing `withFilter` making the ZIO types more uniform.

Comment (jamesward) [reply experiments/src/main/scala/HelloZio/CalculatorExample.scala:117]: Not sure why destructuring the tuple requires the `withFilter`. Could be a Scala bug or unfortunate need.

**Comment by swoogles** [comment]
From the little bit that I understand from Adam's talk, it looks like you made the right updates here.
Thanks for knocking this out!

**Comment by jamesward** [comment]
I had no idea what I was doing but the types guided me. As it should be.

File: experiments/src/main/scala/HelloZio/CalculatorExample.scala (hunk 1 above)
Comment (swoogles) [reply experiments/src/main/scala/HelloZio/CalculatorExample.scala:117]: I've seen chatter about it before on the Scala mailing lists, and it seems to be an unfortunate need.

**Pull request merged**